poo/abstraction/en/Abstraction.md

```diff
@@ -0,0 +1,131 @@
+
+```java
+
+class Hiking {
+
+    List<Human> participants;
+
+    public Hiking() { this.participants = new ArrayList<>(); }
+
+    void addParticipant(Human participant) { this.participants.add(participant); }
+
+    void moveAll() {
+        for(Human human : participants) {
+            human.move();
+            human.consumeEnergy(3);
+        }
+    }
+}
+```
+
+So far so good, we have that we want. This code works perfectly, except, in reality, it contains two major problems.
+
+1. **We restraint the hiking of the objects of type Human**, whereas anything can **walk** and **exhausted** could take a hiking. It’s not really the kind to forbid to elephants and to tractors to participate :(
+2. We ask a whole human but us all we need is their legs (hum). This class hiking, however currently access to the human respiratory system (via `breathe()` ) and to his stomach (via `eat()` ) ! **So, we give too much power to this class Hiking, since it is able to do things who shouldn’t be accessible to it**. <br>
+
+So, how to get around this two problems? 
+
+The **interfaces** come to our rescue (interfaces are great). Let's make a `Walking` interface, which allows to describe something that moves and gets tired:
+
+```java
+interface Walking {
+    void move();
+    void consumeEnergy(float value);
+}
+```
+
+So, now, our class Human will be able to **implement** this super interface, and in reality we add `implements Walking` and put a little `@Override` on their methods `move` and `consumeEnergy` and the turn is played, since this is a behaviour that we had already implemented beforehand, just without the interface.
+
+```java
+class Human implements Walking {
+
+    int air;
+    int satiety;
+    float energy;
+
+
+    void breathe() {
+        air += 10;
+    }
+
+    void eat() {
+        satiety += 5;
+    }
+
+    @Override
+    void move() {
+        System.out.println("Un humain souhaite se déplacer !");
+    }
+
+    @Override
+    void consumeEnergy(float value) {
+        this.energy -= value;
+    }
+}
+```
+
+But, what to change? Now, instead of making a human list in our Hiking, we will do a **Walking list** (it’s more politically correct). And, there is almost nothing to change, it’s a good sign:
+
+```java
+
+class Hiking {
+
+    List<Walking> participants;
+
+    public Hiking() { this.participants = new ArrayList<>(); }
+
+    void addParticipant(Walking participant) { this.participants.add(participant); }
+
+    void moveAll() {
+        for(Walking walking: participants) {
+            walking.move();
+            walking.consumeEnergy(3);
+        }
+    }
+}
+```
+
+And, henceforth, not only our Hiking class hasn’t access to the Human methods `breathe` and `eat`, since, nowhere, we said to our work with humans (no, but think, if a hiking can make you swallow peas, it would be creepy, still), but more, it accepts the elephants and the tractors !

```

Comment:
```suggestion
And, henceforth, not only doesn’t our Hiking class have access to the Human methods `breathe` and `eat`, since, we never specified that it's humans we're working with (think about it, if hiking could make you swallow peas, it would be creepy, right?), but also it accepts the elephants and the tractors!

```
`not only ... but also` has a pretty nasty syntax, so I corrected the few mistakes you made here.